gamemodes/terrortown/gamemode/server/sv_eventpopup.lua

```diff
@@ -0,0 +1,48 @@
+util.AddNetworkString("ttt2_eventpopup")
+
+EPOP = EPOP or {}
+
+---
+-- Adds a translated popup message to the @{EPOP}
+-- @param table|Player plys A table of player that should receive this popup, broadcasts it if nil
+-- @param string|table title The title of the popup that will be displayed in large letters (can be a table with `text` and `color` attribute)
+-- @param [opt]string|table subtitle An optional description that will be displayed below the title (can be a table with `text` and `color` attribute)
+-- @param [default=4]number displayTime The render duration of the popup
+-- @param [default=true]boolean blocking If this is false, this message gets instantly replaced if a new message is added
+-- @realm server
+function EPOP:AddMessage(plys, title, subtitle, displayTime, blocking)
+	net.Start("ttt2_eventpopup")
+
+	if title or isstring(title) then
+		net.WriteBool(false)
+		net.WriteString(title or "")
+	elseif not IsValid(title.color) then
```

Comment:
Here, title could be invalid, This would lead to `nil` error

Reply:
And can a color be valid? Why not using `iscolor` ?

Reply:
you're right here, it should have been `if not title or isstring(title)`

Reply:
Oh okay. A completely changed logic. Then you can encapsulate the `isstring` / istable check and do a simple if-else with title inside or smth